src/Tasks/Microsoft.NET.Build.Tasks/ProjectContext.cs

```diff
@@ -79,7 +79,12 @@ public RuntimeFramework(ITaskItem item)
             }
             else
             {
-                CompilationLockFileTarget = lockFile.GetTargetAndThrowIfNotFound(lockFileTarget.TargetFramework, null);
+                var frameworkAlias = lockFile.PackageSpec.TargetFrameworks.FirstOrDefault(tfi => tfi.FrameworkName == lockFileTarget.TargetFramework).TargetAlias;
```

Comment:
I think it'd nice if you didn't have this dependency, but don't think the current NuGet api has an alternative for you. 
The lockfiletarget can certainly carry the alias...maybe I'll jsut add that. 

You need a null check here: 

```suggestion
                var frameworkAlias = lockFile.PackageSpec.TargetFrameworks.FirstOrDefault(tfi => tfi.FrameworkName == lockFileTarget.TargetFramework)?.TargetAlias;
```